File: tournament/submissions/accepted/tournament.py
def solve(N: int, C: list, P: list) -> str:
    def helper(number, competitors, powers):
        if (number == 1):
            return competitors[0]
        winners = []
        winnerPowers = []
        for i in range(0, number, 2):
            if (powers[i] > powers[i + 1]):
                winners.append(competitors[i])
            elif (powers[i] == powers[i + 1]):
                winners.append(competitors[i] + competitors[i + 1])
            else:    
                winners.append(competitors[i + 1])
            winnerPowers.append(powers[i] + powers[i + 1])
        return helper(number // 2, winners, winnerPowers)
    return helper(N, C, P)


def main():
    T = int(input())
    for _ in range(T):
        N = int(input())
        C = input().split(" ")
        P = list(map(lambda s: int(s), input().split(" ")))
        print(solve(N, C, P))

if __name__ == '__main__':
    main()

File: tournament/submissions/accepted/tournament.java
import java.io.*;
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;

class Solution {
    /**
     * Return a single string of the champion's name
     *
     * N: The length of C and P
     * C: List of strings of the competitors
     * P: List of integers of competitor's power
     */
    static String solve(int N, String[] C, Integer[] P) {
        return helper(Arrays.asList(C), Arrays.asList(P));
    }

    private static String helper(List<String> C, List<Integer> P) {
        if (C.size() == 1) {
            return C.get(0);
        }
        List<String> Winners = new ArrayList<>();
        List<Integer> WinnerPowers = new ArrayList<>();
        for (int i = 0; i < C.size(); i += 2) {
            if (P.get(i) > P.get(i + 1)) {
                Winners.add(C.get(i));
            } else if (P.get(i) == P.get(i + 1)) {
                Winners.add(C.get(i) + C.get(i + 1));
            } else {
                Winners.add(C.get(i + 1));
            }
            WinnerPowers.add(P.get(i) + P.get(i + 1));
        }
        return helper(Winners, WinnerPowers);
    }
    
    static BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    static PrintWriter out = new PrintWriter(System.out);

    public static void main(String[] args) throws IOException {
        int T = Integer.parseInt(in.readLine());
        for (int i = 0; i < T; i++) {
            int N = Integer.parseInt(in.readLine());
            String[] C = in.readLine().split(" ");
            String[] temp = in.readLine().split(" ");
            Integer[] P = new Integer[N];
            for (int j = 0; j < N; j++) {
                P[j] = Integer.parseInt(temp[j]);
            }
            out.println(solve(N, C, P));
        }
        out.flush();
    }
}
